File: it355/it355-pz01-nikola_tasic_3698/src/main/java/com/example/backend/entity/Post.java
package com.example.backend.entity;

import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.List;

/**
 * Blog post
 */
@Data
@Entity
@Table(name = "post")
@EqualsAndHashCode(callSuper = false, onlyExplicitlyIncluded = true)
public class Post extends Auditable {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@EqualsAndHashCode.Include
	@Column(name = "post_id")
	private Integer id;
	@JoinColumn(name = "user_fk", referencedColumnName = "user_id")
	@ManyToOne
	private User user;
	@JoinColumn(name = "category_fk", referencedColumnName = "category_id")
	@ManyToOne
	private Category category;
	@Column(name = "title")
	private String title;
	@Column(name = "excerpt")
	private String excerpt;
	@Column(name = "body")
	private String body;
	@Column(name = "deleted")
	private Boolean deleted;
	@Column(name = "published")
	private Boolean published;
	@Column(name = "views")
	private Integer views;
	@Column(name = "slug")
	private String slug;
	@ManyToMany
	@JoinTable(name = "post_tag", joinColumns = @JoinColumn(name = "post_fk"), inverseJoinColumns = @JoinColumn(name = "tag_fk"))
	private List<Tag> tags;
}

File: cs322/cs322-pz-nikola_tasic_3698/blog-backend/Data/Entity/Post.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;

namespace blog_backend.Data;

[Table("post")]
public class Post : Auditable {
	[Key, Column("post_id")]
	public int Id { get; set; }

	[Column("title")]
	public string Title { get; set; }

	[Column("body")]
	public string Body { get; set; }

	[Column("excerpt")]
	public string Excerpt { get; set; }

	[Column("slug")]
	public string Slug { get; set; }

	[Column("image")]
	public string? Image { get; set; }

	[Column("views")]
	public int? Views { get; set; }

	[Column("category_fk")]
	[ForeignKey("Category")]
	public int CategoryFk { get; set; }
	
	[ForeignKey("CategoryFk")]
	public virtual Category Category { get; set; }

	[Column("published")]
	public bool Published { get; set; }

	[Column("date_posted")]
	public DateTime? DatePosted { get; set; }
	
	[Column("user_fk")]
	[ForeignKey("User")]
	public int UserFk { get; set; }

	[ForeignKey("UserFk")]
	public virtual User User { get; set; }
}

public class PostModelConfig : IEntityTypeConfiguration<Post> {
	public void Configure(EntityTypeBuilder<Post> builder) {
		builder.HasOne(post => post.User);
	}
}